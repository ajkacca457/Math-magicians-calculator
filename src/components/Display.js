import React from 'react';
import PropTypes from 'prop-types';

const Display = props => {
  const { result } = props;
  return (
    <div className="result">
      <h3>{result || '0'}</h3>
    </div>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: '0',
};

export default Display;
