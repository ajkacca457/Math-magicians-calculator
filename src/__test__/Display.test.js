import React from 'react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Display from '../components/Display';

describe('Display', () => {
  test('Correctly renders Display component', () => {
    const displayTree = TestRenderer.create(<Display />).toJSON();
    expect(displayTree).toMatchSnapshot();
  });
});
