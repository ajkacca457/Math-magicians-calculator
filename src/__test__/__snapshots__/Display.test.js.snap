// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Correctly renders Display component 1`] = `
<div
  className="result"
>
  <h3>
    0
  </h3>
</div>
`;
